kernel/kernel-api/src/main/java/io/delta/kernel/TransactionBuilder.java

```diff
@@ -34,11 +34,26 @@
 @Evolving
 public interface TransactionBuilder {
   /**
-   * Set the schema of the table when creating a new table.
+   * Set the schema of the table. If setting the schema on an existing table for a schema evolution,
+   * then column mapping must be enabled. If a schema evolution is performed, implementations must
+   * perform the following validations:
+   *
+   * <ul>
+   *   <li>No duplicate columns are allowed
```

Comment:
do we want to mention type of schema evolution supported: like adding, renaming or dropping a column.

Reply:
Sure I included these details.